server/src/main/java/org/opensearch/cluster/metadata/Metadata.java

```diff
@@ -1391,7 +1391,7 @@ public Builder put(final QueryGroup queryGroup) {
             return queryGroups(existing);
         }
 
-        private Map<String, QueryGroup> getQueryGroups() {
+        public Map<String, QueryGroup> getQueryGroups() {
```

Comment:
Stub, this will be introduced into main by this PR - https://github.com/opensearch-project/OpenSearch/pull/14680/files#diff-27d652d5171910d46ea8d667ae5c44ff6ec70f2f83d9f319adcc3068b8c05ec0R856